<!-- Launch File for running svo with ceres backend from bag-file -->
<launch>
  
  <arg name="type" default="vio"/>
  
  <arg name="calib_file" default="$(find svo_ros)/param/calib/airsim/airsim_triple_with_depth_imu.yaml"/>

  <arg name="config_file" default="$(find svo_ros)/param/airsim_$(arg type)_triple_with_depth.yaml"/>


  <!-- SVO node -->
  <node pkg="svo_ros" type="svo_node" name="svo" clear_params="true" output="screen" args="--v=0">

    <!-- Camera topic to subscribe to -->
    <param name="front_topic" value="/camera/front" type="str" />
    <param name="left_topic" value="/camera/left" type="str" />
    <param name="right_topic" value="/camera/right" type="str" />
    <param name="depth_topic" value="/camera/depth" type="str" />

    <!-- Imu topic to subscribe to -->
    <param name="imu_topic" value="/imu" type="str" />

    <!-- Camera calibration file -->
    <param name="calib_file" value="$(arg calib_file)" />

    <!-- Config file -->
    <param name="config_file" value="$(arg config_file)" />

    <!--Parameters-->
    <param name='runlc' value='false' />

  </node>

  <!-- RVIZ -->
   <node name="vis" pkg="rviz" type="rviz" args=" -d $(find svo_ros)/rviz_config_vio.rviz" />

</launch>
